src/halo_regions.jl

```diff
@@ -24,12 +24,12 @@ _fill_south_halo!(c, ::PBC, H, N) = @views @. c.parent[:, 1:H, :] = c.parent[:,
  _fill_north_halo!(c, ::PBC, H, N) = @views @. c.parent[:, N+H+1:N+2H, :] = c.parent[:, 1+H:2H, :]
 _fill_bottom_halo!(c, ::PBC, H, N) = @views @. c.parent[:, :, N+H+1:N+2H] = c.parent[:, :, 1+H:2H]
```

Comment:
`_fill_top_halo` and `_fill_top_halo` for flux and periodic boundary conditions need to be swapped: the "bottom" grid points range on the parent array from `1:H`, for example.